app/services/xapi/completed_target_service.rb

```diff
@@ -0,0 +1,26 @@
+module Xapi
+  class CompletedTargetService < Xapi::BaseService
```

Comment:
I would prefer composition over inheritance here. I see no big win from having base class here, maybe just include a module with common methods?

Reply:
Naming is hard :) I have some doubts about calling these classes services. They do not encapsulate any behaviour/service. That's just look like a statement's factories for me. After you invoke a constructor you get a xAPI statement. BTW I would model it differently, the xAPI statement build here could be just a value object. So this are statement factories that build xAPI statement and raise error if anything is wrong. Then we should only use build value objects later. Easy to test & use.